analyzers/src/SonarAnalyzer.CFG/LiveVariableAnalysis/RoslynLiveVariableAnalysis.cs

```diff
@@ -161,6 +165,15 @@ private void BuildBranchesFinally(BasicBlock source, ControlFlowRegion finallyRe
         }
     }
 
+    private void BuildBranchesNestedCatchRethrow(BasicBlock block)
+    {
+        var reachableHandlers = block.EnclosingRegion(ControlFlowRegionKind.TryAndCatch).NestedRegion(ControlFlowRegionKind.Try).ReachableHandlers();
```

Comment:
Doesn't `ReachableHandlers` also connect back to it's own `Catch` in this case?

Reply:
@pavel-mikula-sonarsource  I'm after asking for `x.FirstBlockOrdinal > block.Ordinal)` - all the catch with greater ordinal that the one I'm in.
Isn't that working?
```csharp
var reachableHandlers = block.EnclosingRegion(ControlFlowRegionKind.TryAndCatch).NestedRegion(ControlFlowRegionKind.Try).ReachableHandlers();
foreach (var catchBlock in reachableHandlers.Where(x => x.Kind == ControlFlowRegionKind.Catch && x.FirstBlockOrdinal > block.Ordinal).SelectMany(x => x.Blocks(Cfg)))
{}
```

Also not sure what you mean neighbouring catches :/

Reply:
```VB
Try
Catch Ex As IOException
  Throw 'You are here, with your block.Ordinal
Catch Ex As FormatException
   'This is a neighbour that you don't want to connect to. And it has FirstBlockOrdinal > block.Ordinal
End Try
```

Reply:
Go escape, you'd need to store you're `TryAndCatch` region, and be bigger than its `LastBlockOrdinal`

Reply:
Got It! 
I'm opening a PR to fix this.